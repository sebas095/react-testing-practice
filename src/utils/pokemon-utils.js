export const sanitizePokemonId = (pokemonId) => {
  let sanitizedId = pokemonId;

  if (pokemonId < 10) sanitizedId = `00${pokemonId}`;
  else if (pokemonId < 100) sanitizedId = `0${pokemonId}`;

  return sanitizedId;
};

export const getPokemonImage = (pokemonId) => {
  return `https://assets.pokemon.com/assets/cms2/img/pokedex/detail/${pokemonId}.png`;
};
