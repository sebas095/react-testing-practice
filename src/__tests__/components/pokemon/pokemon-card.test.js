import { shallow } from "enzyme";

import PokemonCard from "../../../components/pokemon/pokemon-card";
import {
  PokemonName,
  PokemonImage,
} from "../../../components/pokemon/pokemon-styled";

import { sanitizePokemonId } from "../../../utils/pokemon-utils";

const name = "Bulbasaur";
const url = "https://pokeapi.co/api/v2/pokemon/1";

let pokemonId = Number(url.split("/")[6]);
pokemonId = sanitizePokemonId(pokemonId);

describe("Pokemon Card", () => {
  it("Render ok", () => {
    const wrapper = shallow(<PokemonCard name={name} pokemonId={pokemonId} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Show Pokemon Name component", () => {
    const wrapper = shallow(<PokemonCard name={name} pokemonId={pokemonId} />);
    expect(wrapper.find(PokemonName)).toHaveLength(1);
    expect(wrapper.find(PokemonName).text()).toEqual(`${pokemonId}: ${name}`);
  });

  it("Show Pokemon Image component", () => {
    const wrapper = shallow(<PokemonCard name={name} pokemonId={pokemonId} />);
    expect(wrapper.find(PokemonImage)).toHaveLength(1);
  });
});
