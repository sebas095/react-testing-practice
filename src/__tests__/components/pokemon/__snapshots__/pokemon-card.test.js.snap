// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon Card Render ok 1`] = `ShallowWrapper {}`;
